packages/react-components/src/box/src/Box.tsx

```diff
@@ -0,0 +1,37 @@
+import { AsAttributeWithChildren, ForwardedRefAttribute, SlotAttribute, omitProps } from "../../shared";
+import { ReactElement, forwardRef } from "react";
+
+export interface StrictBoxProps extends AsAttributeWithChildren, SlotAttribute, ForwardedRefAttribute {
+}
+
+export interface BoxProps extends StrictBoxProps {
```

Comment:
Pattern taken from Semantic-UI. 

- React-spectrum : doesn't have as property
- Chakra : doesn't need because they specifiy the as property type explicitly. Smart, but requires more code than we would like to make it work
- React-bootstrap : extends React.HTMLAttributes, which in my opinion is misleading since we allow custom components, and that component might not expect those properties. I think i'd rather have an untyped property bag than misleading props